sources/packages/backend/apps/api/src/route-controllers/student-appeal/student-appeal.students.controller.ts

```diff
@@ -161,6 +166,18 @@ export class StudentAppealStudentsController extends BaseController {
       userToken.userId,
       appealRequests,
     );
+
+    const student = await this.studentService.getStudentByUserId(
+      userToken.userId,
+    );
+
+    await this.studentFileService.updateStudentFiles(
+      student.id,
+      userToken.userId,
+      payload.studentAppealRequests[0].files,
```

Comment:
An appeal can be submitted with multiple appeals (financial info, pd status and etc etc)

We cannot assume that first one will have files. 

You can create a flatmap of all the files and call update Student files and this can be done inside the service.